### PR TITLE
[ISSUE-812] Support GA versions in release workflow

### LINKED ISSUE #812: Improve release workflow to support GA releases
**Is your feature request related to a problem? Please describe.**
Currently, release workflow allows only pre-releases creation. It's needed to be able to create GA release as well.

**Describe the solution you'd like**
Add prerelease parameter obtained from GitHub release object and passed to workflow in operator repo.
In operator repo determine is it release or pre-release by parameter value.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by misanthropicat**
### Resolves #812 

Added passing prerelease parameter to release workflow in operator repo.
Added targets for release workflow linting and local testing.
Updated doc.

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
[Run on fork](https://github.com/misanthropicat/csi-baremetal/actions/runs/2171670661)

### CODE REVIEW

**Review by mitsaa — APPROVED** [review]

**Review by rkiselenko — APPROVED** [review]

File: docs/release-workflow.md [hunk 1]
```diff
@@ -7,14 +7,34 @@ Reduce manually performed operation during csi release procedure.
 Use GitHub actions for release charts and update helm repository.
 
 Whole release workflow contains 2 parts:
-1. Release workflow in current repo triggered on release creation
+1. Release workflow in current repo triggered on release or pre-release creation
 2. Release workflow in [csi-baremetal-operator](https://github.com/dell/csi-baremetal-operator) repo
 
-### Detailed description
-
+## Detailed description
```
Comment (mishoyama) [inline docs/release-workflow.md:13]: Please remove Proposal word since this is documentation now. What steps should I do to release CSI charts?

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: Makefile.validation [hunk 2]
```diff
@@ -50,6 +50,19 @@ test-sanity:
 #	should fail when requesting to create a volume with already existing name and different capacity|\
 #	should not fail when requesting to create a volume with already existing name and same capacity" -ginkgo.v -timeout=0
 
+### Workflows
+
+prepare-env:
+	go install github.com/rhysd/actionlint/cmd/actionlint@latest
+	go install github.com/nektos/act@latest
+
+workflows-lint:
```
Comment (safronovD) [inline Makefile.validation:59]: Could we make this target a part of PR validation and run it for all workflows?

File: docs/release-workflow.md [hunk 3]
```diff
@@ -7,14 +7,34 @@ Reduce manually performed operation during csi release procedure.
...
+1. `make prepare-env` will download and install mentioned tools
+2. `make workflows-lint` will lint all workflows in `.github/workflows` dir
+3. Create file `.github/workflows/tests/wf.secrets` with GITHUB_TOKEN variable definition. Token is generated on [token page of GitHub user settings](https://github.com/settings/tokens). Required permissions: repo, workflow.
+4. Login to ghcr.io with GH username and GITHUB_TOKEN: `echo $(cat .github/workflows/tests/wf.secrets | awk -F "=" '{print $2}') | docker login ghcr.io -u USERNAME --password-stdin`
+5. `make test-release-workflow` will run workflow locally and execute all steps except marked as `${{ !env.ACT }}`
+
+Release workflow in csi-baremetal-operator workflow test is started by the following cmd:
+`make test-release-workflow`
+
+## Usage
```
Comment (safronovD) [inline docs/release-workflow.md:35]: Do you mean usage for test purpose or for release creating?

File: docs/release-workflow.md [hunk 4]
```diff
@@ -7,14 +7,34 @@ Reduce manually performed operation during csi release procedure.
...
+Create release in this repo choosing necessary tag, target, fill title and description using the following format:
```
Comment (safronovD) [inline docs/release-workflow.md:36]: As I understand it, we have the following parameters to run action:
1. tag
2. branch
3. is_prerelease
4. csi/operator version

Could we find branch name and csi/operator versions by tag and reduce these parameters?

File: Makefile.validation (hunk 2 above)
Comment (misanthropicat) [reply Makefile.validation:59]: Actually, this target will run for all workflows located in .github/workflows dir.

File: docs/release-workflow.md (hunk 3 above)
Comment (misanthropicat) [reply docs/release-workflow.md]: Usage is for release creating, I've just added some details to doc, probably it became more clear :)

File: docs/release-workflow.md (hunk 4 above)
Comment (misanthropicat) [reply docs/release-workflow.md]: We can't go without tag and branch, as GH won't let you sending a form without these parameters. Is_prerelease is necessary for deciding which release type should be created in operator repo, as it's the same as in this repo. As for versions - yes, we'd be able to reduce them after the next workflow improvement about test coverage. We discussed it today.

File: docs/release-workflow.md (hunk 1 above)
Comment (misanthropicat) [reply docs/release-workflow.md]: Done. I've clarified 'Usage' doc section, does it answer your question?

File: Makefile.validation (hunk 2 above)
Comment (safronovD) [reply Makefile.validation:59]: Excellent! May I ask you to try to add this target to PR validation workflow?

File: docs/release-workflow.md [hunk 5]
```diff
@@ -3,18 +3,35 @@
...
+3. Create file `.github/workflows/tests/wf.secrets` with GITHUB_TOKEN variable definition. Token is generated on [token page of GitHub user settings](https://github.com/settings/tokens). Required permissions: repo, workflow.
+4. Login to ghcr.io with GH username and GITHUB_TOKEN: `echo $(cat .github/workflows/tests/wf.secrets | awk -F "=" '{print $2}') | docker login ghcr.io -u USERNAME --password-stdin`
+5. `make test-release-workflow` will run workflow locally and execute all steps except marked as `${{ !env.ACT }}`
+
+Release workflow in csi-baremetal-operator workflow test is started by the following cmd in csi-baremetal-operator dir:
+`make test-release-workflow`
+
+## Workflow usage
+Steps to create release with all necessary artefacts, changelog and update gh-pages with release-workflow:
+1. Manually create release in this repo choosing necessary tag, target branch, fill release title and description using the following format:
```
Comment (safronovD) [inline docs/release-workflow.md:34]: Should we tag choosen for release commits at first?

Comment (safronovD) [inline docs/release-workflow.md:34]: In this or in the operator repo?

Comment (misanthropicat) [reply docs/release-workflow.md:34]: In this repo. Release in operator repo is created by [workflow](https://github.com/dell/csi-baremetal-operator/blob/4dd5bfcf4e4b4b82858265bb506f55df9a722c65/.github/workflows/release.yml#L105)

File: docs/release-workflow.md [hunk 6]
```diff
@@ -3,18 +3,35 @@
...
+2. `make workflows-lint` will lint ==all== workflows in `.github/workflows` dir. It's possible to lint certain workflow (as described [here](https://github.com/rhysd/actionlint/blob/main/docs/usage.md#actionlint-command)), but I found better to lint all at once as it's fast enough.
+3. Create file `.github/workflows/tests/wf.secrets` with GITHUB_TOKEN variable definition. Token is generated on [token page of GitHub user settings](https://github.com/settings/tokens). Required permissions: repo, workflow.
+4. Login to ghcr.io with GH username and GITHUB_TOKEN: `echo $(cat .github/workflows/tests/wf.secrets | awk -F "=" '{print $2}') | docker login ghcr.io -u USERNAME --password-stdin`
+5. `make test-release-workflow` will run workflow locally and execute all steps except marked as `${{ !env.ACT }}`
+
+Release workflow in csi-baremetal-operator workflow test is started by the following cmd in csi-baremetal-operator dir:
+`make test-release-workflow`
+
+## Workflow usage
+Steps to create release with all necessary artefacts, changelog and update gh-pages with release-workflow:
```
Comment (mishoyama) [inline docs/release-workflow.md:33]: typo `artefacts` ->  `artifacts`

**Review by mishoyama — APPROVED** [review]

File: docs/release-workflow.md (hunk 6 above)
Comment (misanthropicat) [reply docs/release-workflow.md:33]: Both variants are [valid](https://translate.google.com/?hl=ru&tab=TT&sl=ru&tl=en&text=%D0%B0%D1%80%D1%82%D0%B5%D1%84%D0%B0%D0%BA%D1%82&op=translate): 
image

File: Makefile.validation (hunk 2 above)
Comment (misanthropicat) [reply Makefile.validation:59]: Do you mean separate target only for PR validation?

**Pull request merged**